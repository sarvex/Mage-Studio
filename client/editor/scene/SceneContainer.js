import React from 'react';
import {
    connect
} from 'react-redux';

import classnames from 'classnames';

// import {
//     meshChanged,
//     meshAttached,
//     meshDetached
// } from '../../actions/currentMesh';

import { elementAttached, toggleInspectorVisibility } from '../../actions/inspector';

import {
    saveScene,
    loadScene,
    startProject,
    stopProject
} from '../../actions/scene';

import Scene from './Scene';
// import Player from './Player';

import style from './scene.module.scss';
// import { showModelUploadModal } from '../../actions/models';

export class SceneContainer extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            fullscreen: false
        };
    }

    handleFullScreen = (fullscreen) => {
        this.setState({ fullscreen });
    }

    render() {
        const {
            scene,
            store,
            config,
            loadScene,
            onElementChanged,
            onElementAttached,
            onElementDetached,
            onSceneExported,
            startProject,
            showModelModal,
            stopProject
        } = this.props;

        const { fullscreen } = this.state;
        // const { projectPlayerVisible, projectUrl } = scene;

        const className = classnames(style['scene-container'], {
            [style.fullscreen]: fullscreen
        });

        return (
            <div className={className}>
                {/* <Player
                    url={projectUrl}
                    visible={projectPlayerVisible} /> */}
                <Scene
                    fullscreen={fullscreen}
                    config={config}
                    store={store}
                    onSceneLoad={loadScene}
                    onElementChanged={onElementChanged}
                    onElementAttached={onElementAttached}
                    onElementDetached={onElementDetached}
                    onSceneExported={onSceneExported} />
            </div>
        );
    }
}

const mapStateToProps = ({ config, scene }) => ({
    // config,
    // scene
});

const mapDispatchToProps = (dispatch) => ({
    // startProject: project => dispatch(startProject(project)),
    // showModelModal: () => dispatch(showModelUploadModal()),
    // onElementChanged: ({name, position, rotation, scale}) => {
    //     return dispatch(meshChanged(name, position, rotation, scale))
    // },
    onElementAttached: () => dispatch(elementAttached()),
    onElementDetached: () => dispatch(toggleInspectorVisibility(false)),
    // onSceneExported: (name) => ({ data }) => dispatch(saveScene(name, data)),
    // loadScene: (name) => dispatch(loadScene(name)),
    // stopProject: (project) => dispatch(stopProject(project))
});

export default connect(mapStateToProps, mapDispatchToProps)(SceneContainer);
