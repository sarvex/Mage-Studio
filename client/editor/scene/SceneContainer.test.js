import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import sinon from 'sinon';
import { SceneContainer } from './SceneContainer';

const getSceneContainer = (props) => shallow(<SceneContainer scene={{}} {...props} />);

describe('SceneContainer', () => {

    it('should render properly', () => {
        const component = getSceneContainer();

        expect(toJSON(component)).toMatchSnapshot();
    });

    it('should render properly if fullscreen', () => {
       const component = getSceneContainer();

       component.setState({ fullscreen: true });

       expect(toJSON(component)).toMatchSnapshot();
    });

    it('should have state.fullscreen set to false', () => {
        const component = getSceneContainer();

        expect(component.state().fullscreen).toBe(false);
    });

    describe('events', () => {

        it('should call onElementChanged', () => {
            const spy = sinon.spy();
            const component = getSceneContainer({ onElementChanged: spy });

            component.find('Scene').simulate('elementChanged');

            expect(spy.called).toBe(true);
        });

        it('should call onElementAttached', () => {
            const spy = sinon.spy();
            const component = getSceneContainer({ onElementAttached: spy });

            component.find('Scene').simulate('elementAttached');

            expect(spy.called).toBe(true);
        });

        it('should call onElementDetached', () => {
            const spy = sinon.spy();
            const component = getSceneContainer({ onElementDetached: spy });

            component.find('Scene').simulate('elementDetached');

            expect(spy.called).toBe(true);
        });

        it('should call onSceneExported', () => {
            const spy = sinon.spy();
            const component = getSceneContainer({ onSceneExported: spy });

            component.find('Scene').simulate('sceneExported');

            expect(spy.called).toBe(true);
        });
    });
});