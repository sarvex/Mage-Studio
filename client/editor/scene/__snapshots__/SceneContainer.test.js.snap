// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SceneContainer should render properly 1`] = `
<div
  className="scene-container"
>
  <Scene
    fullscreen={false}
  />
</div>
`;

exports[`SceneContainer should render properly if fullscreen 1`] = `
<div
  className="scene-container fullscreen"
>
  <Scene
    fullscreen={true}
  />
</div>
`;
